lib/rubocop/cop/style/class_structure.rb

```diff
@@ -0,0 +1,312 @@
+      #
+      #     # followed by other macros (if any)
+      #     validates :name
+      #
+      #     # public class methods are next in line
+      #     def self.some_method
+      #     end
+      #
+      #     # initialization goes between class methods and instance methods
+      #     def initialize
+      #     end
+      #
+      #     # followed by other public instance methods
+      #     def some_method
+      #     end
+      #
+      #     # protected and private methods are grouped near the end
+      #     protected
+      #
+      #     def some_protected_method
+      #     end
+      #
+      #     private
+      #
+      #     def some_private_method
+      #     end
+      #   end
+      #
+      # @see https://github.com/bbatsov/ruby-style-guide#consistent-classes
+      class ClassStructure < Cop
+        HUMANIZED_NODE_TYPE = {
+          casgn: :constant,
+          defs: :class_method,
+          def: :instance_method
+        }.freeze
+
+        VISIBILITIES = %i[private protected public].freeze
+        MSG = '`%s` is supposed to appear before `%s` in the class definition.'.freeze
+
+        # Validates code style on class declaration.
+        # Add offense when find a node out of expected order.
+        def on_class(class_node)
+          previous = -1
+          walk_over_nested_class_definition(class_node) do |node, category|
+            index = expected_order.index(category)
+            if index < previous
+              msg = format(MSG, category, expected_order[previous])
+              add_offense(node, location: :expression, message: msg)
+            end
+            previous = index
+          end
+        end
+
+        # Autocorrect by swapping between two nodes autocorrecting them
+        def autocorrect(node)
+          node_classification = classify(node)
+          previous = left_siblings_of(node).find do |sibling|
+            classification = classify(sibling)
+            !ignore?(classification) && node_classification != classification
+          end
+
+          current_range = source_range_with_comment(node)
+          previous_range = source_range_with_comment(previous)
+
+          lambda do |corrector|
+            corrector.insert_before(previous_range, current_range.source)
+            corrector.remove(current_range)
+          end
+        end
+
+        private
+
+        # Classifies a node to match with something in the {expected_order}
+        # @param node to be analysed
+        # @return String when the node type is a `:block` then
+        #   {classify} recursively with the first children
+        # @return String when the node type is a `:send` then {find_category}
+        #   by method name
+        # @return String otherwise trying to {humanize_node} of the current node
+        def classify(node)
+          return node.to_s unless node.respond_to?(:type)
+          case node.type
+          when :block
+            classify(node.children.first)
```

Comment:
First child of a block node is `send_node`, I believe. So you should be able to use `node.send_node`. 🙂

Reply:
Nice! I didn't know about this method 🎉